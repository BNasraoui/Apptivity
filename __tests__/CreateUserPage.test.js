/**
 * @format
 */

import 'react-native';
import React from 'react';
import CreateUserPage from '../screens/CreateUserPage';
import { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

// Snapshot testing when the screen is initially loaded. Run with -u to update snapshot if changes are made
test('inital screen snapshot testing', () => {
	const screen = shallow(<CreateUserPage />);
	expect(toJson(screen)).toMatchSnapshot();
});

// Testing Study ID Components
describe('Study ID tests', () => {
	let screen = shallow(<CreateUserPage />);

	// Testing text is correct
	test('Study ID text', () => {
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'StudyIDView');
		// Check the text
		expect(component.children('BaseText').prop('children')).toEqual("ID:");
	 });

	// Testing text input is updating state correctly
	test('Input field updating', () => {
		// Default state value is empty
		expect(screen.state('inputID')).toEqual("");
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'StudyIDView');
		// Find TextInput component
		let textInput = component.children('TextInput');
		// Change the text
		textInput.props().onChangeText("test123");
		// Update screen
		screen.update();
		// Check state is updated correctly
		expect(screen.state('inputID')).toEqual("test123");
	 });

});

// Testing Epoch Components
describe('Epoch tests', () => {
	
	// Testing text is correct
	test('Epoch text', () => {
		let screen = shallow(<CreateUserPage />);
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DiaryEpochView');
		// Check the text
		expect(component.children('BaseText').prop('children')).toEqual("Diary Epoch:");
	 });

	// Testing picking 5 mins on epoch is working correctly
	test('Picker five mins', () => {
		let screen = shallow(<CreateUserPage />);
		// Default state value is empty
		expect(screen.state('epoch')).toEqual("");
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DiaryEpochView');
		// Find Picker component
		let picker = component.children('View').children('Picker');
		// Find required Picker.Item 
		epochValue = picker.childAt(0).prop('value');
		// Change the picker option
		picker.props().onValueChange(epochValue, 0);
		// Update screen
		screen.update();
		// Check state is updated correctly
		expect(screen.state('epoch')).toEqual("5");
	 });

	// Testing picking 15 mins on epoch is working correctly
	test('Picker fifteen mins', () => {
		let screen = shallow(<CreateUserPage />);
		// Default state value is empty
		expect(screen.state('epoch')).toEqual("");
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DiaryEpochView');
		// Find Picker component
		let picker = component.children('View').children('Picker');
		// Find required Picker.Item 
		epochValue = picker.childAt(1).prop('value');
		// Change the picker option
		picker.props().onValueChange(epochValue, 1);
		// Update screen
		screen.update();
		// Check state is updated correctly
		expect(screen.state('epoch')).toEqual("15");
	 });

	// Testing picking 30 mins on epoch is working correctly
	test('Picker thirty mins', () => {
		let screen = shallow(<CreateUserPage />);
		// Default state value is empty
		expect(screen.state('epoch')).toEqual("");
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DiaryEpochView');
		// Find Picker component
		let picker = component.children('View').children('Picker');
		// Find required Picker.Item 
		epochValue = picker.childAt(2).prop('value');
		// Change the picker option
		picker.props().onValueChange(epochValue, 2);
		// Update screen
		screen.update();
		// Check state is updated correctly
		expect(screen.state('epoch')).toEqual("30");
	 });

});

// Testing adding personalised activities
describe('Adding personalised activities', () => {
	
	// Testing initial value of personalisedActivities
	test('personalisedActivities initial value', () => {
		let screen = shallow(<CreateUserPage />);
		// Default should be empty
		const expected = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	 });

	test('personalisedActivities adding one activity', () => {
		let screen = shallow(<CreateUserPage />);
		// Check starts as empty
		const expectedEmpty = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expectedEmpty));
		// Find DialogInput
		let component = screen.children('DialogInput');
		// Call DialogInput.submitInput with a test activity to add
		component.props().submitInput('test');
		// Update screen
		screen.update();
		// Check state is updated
		const expected = ['test'];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	});

	test('personalisedActivities adding two activities', () => {
		let screen = shallow(<CreateUserPage />);
		// Check starts as empty
		const expectedEmpty = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expectedEmpty));
		// Find DialogInput
		let component = screen.children('DialogInput');
		// Call DialogInput.submitInput with a test activity to add
		component.props().submitInput('test');
		component.props().submitInput('test2');
		// Update screen
		screen.update();
		// Check state is updated
		const expected = ['test', 'test2'];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	});

	test('personalisedActivities adding three activities', () => {
		let screen = shallow(<CreateUserPage />);
		// Check starts as empty
		const expectedEmpty = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expectedEmpty));
		// Find DialogInput
		let component = screen.children('DialogInput');
		// Call DialogInput.submitInput with a test activity to add
		component.props().submitInput('test');
		component.props().submitInput('test2');
		component.props().submitInput('test3');
		// Update screen
		screen.update();
		// Check state is updated
		const expected = ['test', 'test2', 'test3'];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	});

	test('personalisedActivities limited to three activities', () => {
		let screen = shallow(<CreateUserPage />);
		// Check starts as empty
		const expectedEmpty = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expectedEmpty));
		// Find DialogInput
		let component = screen.children('DialogInput');
		// Call DialogInput.submitInput with a test activity to add
		component.props().submitInput('test');
		component.props().submitInput('test2');
		component.props().submitInput('test3');
		component.props().submitInput('test4');
		// Update screen
		screen.update();
		// Check state is updated. Limit of 3 activities, so 4th activity should not be added.
		const expected = ['test', 'test2', 'test3'];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
		expect(screen.state('personalisedActivities')).not.toContain('test4');
	});
});

// Testing deleting personalised activities
describe('Deleting personalised activities', () => {
	
	// Testing initial value of personalisedActivities
	test('personalisedActivities initial value', () => {
		let screen = shallow(<CreateUserPage />);
		// Default should be empty
		const expected = [];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	 });

	test('personalisedActivities deleting activity', () => {
		let screen = shallow(<CreateUserPage />);
		// Find DialogInput
		let component = screen.children('DialogInput');
		// Call DialogInput.submitInput with a test activity to add
		component.props().submitInput('test');
		component.props().submitInput('test2');
		component.props().submitInput('test3');
		// Update screen
		screen.update();
		// Check state is correct before deleting
		const expected = ['test', 'test2', 'test3'];
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
		// Find the View component containing the required elements
		let componentView = screen.findWhere(node => node.prop('testID') === 'PersonalisedActivitiesContainer');
		// Set index of activity to delete
		deleteIndex = 1;
		// Find one of the activities to delete
		let testView = componentView.findWhere(node => node.key() === String(deleteIndex));
		// Find the name of the activity
		let name = testView.children('PersonalisedActivity').prop('children');
		// Press the delete button
		testView.children('TouchableHighlight').props().onPress();
		// Update screen
		screen.update();
		// Check activity is deleted from the array
		expect(screen.state('personalisedActivities')).not.toContain(name);
		// Update expected array
		expected.splice(deleteIndex, 1);
		// Check previous elements are not deleted from the array
		expect(screen.state('personalisedActivities')).toEqual(expect.arrayContaining(expected));
	});

});

// Testing DataGraph Components
describe('DataGraph Components', () => {
	let screen = shallow(<CreateUserPage />);

	// Testing text is correct
	test('DataGraph Text', () => {
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DataGraphView');
		// Check the text
		expect(component.children('BaseText').prop('children')).toEqual("Data Graph:");
	 });

	// Testing Switch initial value
	test('DataGraph Switch initial value', () => {
		// Default state value of screen should be false
		expect(screen.state('switchValueDataGraph')).toEqual(false);
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DataGraphView');
		// Default Switch value should be false
		expect(component.children('Switch').prop('value')).toEqual(false);
	 });

	// Testing Switch changing value
	test('toggleSwitchDataGraph change value', () => {
		// Find the View component containing the required elements
		let component = screen.findWhere(node => node.prop('testID') === 'DataGraphView');
		// Call the onValueChange with true
		component.children('Switch').props().onValueChange(true);
		// Update the screen
		screen.update();
		// state value of screen should be true
		expect(screen.state('switchValueDataGraph')).toEqual(true);
		// Find the View component containing the required elements
		let componentNew = screen.findWhere(node => node.prop('testID') === 'DataGraphView');
		// Switch value should be true
		expect(componentNew.children('Switch').prop('value')).toEqual(true);
	});
});

// Testing Monitor Position Components
describe('Monitor Position Components', () => {
	let screen = shallow(<CreateUserPage />);

	// Testing Wrist Components
	describe('Wrist Components', () => {
		// Testing text is correct
		test('Wrist Text', () => {
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'WristView');
			// Check the text
			expect(component.children('BaseText').prop('children')).toEqual("Wrist:");
		});

		// Testing Switch initial value
		test('Wrist Switch initial value', () => {
			// Default state value of screen should be false
			expect(screen.state('switchValueWrist')).toEqual(false);
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'WristView');
			// Default Switch value should be false
			expect(component.children('Switch').prop('value')).toEqual(false);
		});

		// Testing Switch changing value
		test('toggleSwitchWrist change value', () => {
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'WristView');
			// Call the onValueChange with true
			component.children('Switch').props().onValueChange(true);
			// Update the screen
			screen.update();
			// state value of screen should be true
			expect(screen.state('switchValueWrist')).toEqual(true);
			// Find the View component containing the required elements
			let componentNew = screen.findWhere(node => node.prop('testID') === 'WristView');
			// Switch value should be true
			expect(componentNew.children('Switch').prop('value')).toEqual(true);
		});
	});

	// Testing Hip Components
	describe('Hip Components', () => {
		// Testing text is correct
		test('Hip Text', () => {
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'HipView');
			// Check the text
			expect(component.children('BaseText').prop('children')).toEqual("Hip:");
		});

		// Testing Switch initial value
		test('Hip Switch initial value', () => {
			// Default state value of screen should be false
			expect(screen.state('switchValueHip')).toEqual(false);
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'HipView');
			// Default Switch value should be false
			expect(component.children('Switch').prop('value')).toEqual(false);
		});

		// Testing Switch changing value
		test('toggleSwitchHip change value', () => {
			// Find the View component containing the required elements
			let component = screen.findWhere(node => node.prop('testID') === 'HipView');
			// Call the onValueChange with true
			component.children('Switch').props().onValueChange(true);
			// Update the screen
			screen.update();
			// state value of screen should be true
			expect(screen.state('switchValueHip')).toEqual(true);
			// Find the View component containing the required elements
			let componentNew = screen.findWhere(node => node.prop('testID') === 'HipView');
			// Switch value should be true
			expect(componentNew.children('Switch').prop('value')).toEqual(true);
		});
	});
});