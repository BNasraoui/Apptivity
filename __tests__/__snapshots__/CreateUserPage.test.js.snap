// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inital screen snapshot testing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2c549e",
      "flex": 1,
      "flexWrap": "wrap",
      "padding": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "white",
        },
      ]
    }
    testID="StudyIDView"
  >
    <BaseText>
      ID:
    </BaseText>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter ID Here"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "width": "75%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "white",
        },
      ]
    }
    testID="DataGraphView"
  >
    <BaseText>
      Data Graph:
    </BaseText>
    <Switch
      onValueChange={[Function]}
      value={false}
    />
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "white",
      }
    }
  >
    <BaseText
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      Monitor Position:
    </BaseText>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
      testID="WristView"
    >
      <BaseText
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        Wrist:
      </BaseText>
      <Switch
        onValueChange={[Function]}
        value={false}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
      testID="HipView"
    >
      <BaseText
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        Hip:
      </BaseText>
      <Switch
        onValueChange={[Function]}
        value={false}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "white",
        },
      ]
    }
    testID="DiaryEpochView"
  >
    <BaseText>
      Diary Epoch:
    </BaseText>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "width": "35%",
        }
      }
    >
      <Picker
        mode="dialog"
        onValueChange={[Function]}
        selectedValue=""
      >
        <PickerItem
          label="5 mins"
          value="5"
        />
        <PickerItem
          label="15 mins"
          value="15"
        />
        <PickerItem
          label="30 mins"
          value="30"
        />
      </Picker>
    </View>
  </View>
  <BaseText
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    Personalised Activity:
  </BaseText>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    testID="PersonalisedActivitiesContainer"
  />
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 20,
        "borderWidth": 1,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
    underlayColor="black"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 40,
          "textAlign": "center",
        }
      }
    >
       + 
    </Text>
  </TouchableHighlight>
  <DialogInput
    closeDialog={[Function]}
    hintInput="Enter Activity Here"
    isDialogVisible={false}
    message="Enter Personalised Activity"
    submitInput={[Function]}
    title="Adding New Personalised Acitvity"
  />
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "flex-end",
        "alignSelf": "flex-end",
        "backgroundColor": "white",
        "borderRadius": 30,
        "borderWidth": 1,
        "height": 60,
        "justifyContent": "center",
        "marginRight": 20,
        "width": 60,
      }
    }
    underlayColor="black"
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/tick.png",
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </TouchableHighlight>
</View>
`;
